modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+  if (!is.null(projection)) {
+    PEcAn.logger::severeifnot("projection should be of character type",
+                              is.character(projection))
+  }
+  if (!is.null(algorithm)) {
+    PEcAn.logger::severeifnot("algorithm should be of character type",
+                              is.character(algorithm))
+  }
+  if (!is.null(credfile)) {
+    PEcAn.logger::severeifnot("credfile should be of character type",
+                              is.character(credfile))
+  }
+  PEcAn.logger::severeifnot(
+    "Check if out_get_data is of character type and is not NULL",
+    is.character(out_get_data)
+  )
+  if (!is.null(out_process_data)) {
+    PEcAn.logger::severeifnot("out_process_data should be of character type",
+                              is.character(out_process_data))
+  }
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(
+    stringsAsFactors = FALSE,
+    original_name = c(
+      "LANDSAT/LC08/C01/T1_SR",
+      "COPERNICUS/S2_SR",
+      "NASA_USDA/HSL/SMAP_soil_moisture"
+    ),
+    pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
+  
+  if (source == "gee") {
+    collection = unname(getpecancode[collection])
+  }
+  
+  # construct raw file name
+  raw_file_name <-
+    construct_raw_filename(collection, siteid_short, scale, projection, qc)
+  
+  
+  # check if any data is already present in the inputs table
+  dbstatus <-
+    remotedata_db_check(
+      raw_file_name     = raw_file_name,
+      start             = start,
+      end               = end,
+      siteid            = siteid,
+      siteid_short      = siteid_short,
+      out_get_data      = out_get_data,
+      algorithm         = algorithm,
+      out_process_data  = out_process_data,
+      overwrite         = overwrite,
+      dbcon             = dbcon
+    )
+  
+  remotefile_check_flag  <- dbstatus[[1]]
+  start                  <- dbstatus[[2]]
+  end                    <- dbstatus[[3]]
+  stage_get_data         <- dbstatus[[4]]
+  write_raw_start        <- dbstatus[[5]]
+  write_raw_end          <- dbstatus[[6]]
+  raw_merge              <- dbstatus[[7]]
+  existing_raw_file_path <- dbstatus[[8]]
+  stage_process_data     <- dbstatus[[9]]
+  write_pro_start        <- dbstatus[[10]]
+  write_pro_end          <- dbstatus[[11]]
+  pro_merge              <- dbstatus[[12]]
+  input_file             <- dbstatus[[13]]
+  existing_pro_file_path <- dbstatus[[14]]
+  raw_check              <- dbstatus[[15]]
+  pro_check              <- dbstatus[[16]]
+  
+  # construct outdir path
+  outdir <-
+    file.path(outdir, paste(source, "site", siteid_short, sep = "_"))
+
+  # extract the AOI of the site from BETYdb
+  coords <-
```

Comment:
like we talked, it would be great if you could add the example of storing polygon in the DB to the documentation

```
db.query("insert into sites (country, sitename, geometry) values ('FI', 'Qvidja_ca6cm', ST_SetSRID(ST_MakePolygon(ST_GeomFromText('LINESTRING(22.388957339620813 60.287395608412218 14.503780364990234, 22.389600591651835 60.287182336733203 14.503780364990234,
22.38705422266651  60.285516177775868 14.503780364990234,      
22.386575219445195 60.285763643883932 14.503780364990234,
22.388957339620813 60.287395608412218 14.503780364990234 )')), 4326));", con = dbcon)
```

If the site doesn't have the polygon it would return a point, e.g.
```

>   coords <-
+     unlist(PEcAn.DB::db.query(
+       sprintf("select ST_AsGeoJSON(geometry) from sites where id=%f", 1000000758),
+       con = dbcon
+     ), use.names = FALSE)
> coords
[1] "{\"type\":\"Point\",\"coordinates\":[-95.2001768,30.1625267,0]}"
```

Would we want to check if the downstream functions need a polygon type or not here?

Reply:
> If the site doesn't have the polygon it would return a point, e.g.

Do you mean adding documentation for creating Point site? Because otherwise as you showed in your bottom example this query does return geojson info for Point site as well.

> Would we want to check if the downstream functions need a polygon type or not here?

These are being taken care of by the functions in `RpTools`

Reply:
>Do you mean adding documentation for creating Point site?

No no for creating the polygon site, vast majority of the sites on Bety DB has point. But a lot of people could want to use your functions with the polygon

>These are being taken care of by the functions in RpTools

OK, but it could be harder for people to debug there, if the issue is as simple as forgetting to create the polygon for the site, we might as well let them know here. I fell like this will be a common issue just because we are not used to create these polygons for the sites. We can put this info into the registry files as well and put a high level warning here, e.g. assume these are the tags in the registry.xml:

```
<coord>
  <coordtype>polygon</coordtype>
</coord>
```
If the code above returns point, we throw a warning (or an error if we are sure it will fail?)

This could be the registry for sources/collections that can work with both types:
```
<coord>
  <coordtype>polygon</coordtype>
  <coordtype>point</coordtype>
</coord>
```